pkg/core/rest/errors/error_handler.go

```diff
@@ -130,7 +98,7 @@ func HandleError(ctx context.Context, response *restful.Response, err error, tit
 			Title:  "Method not Allowed",
 			Detail: err.Error(),
 		}
-	case errors.Is(err, &Conflict{}):
+	case errors.Is(err, &Conflict{}) || errors.Is(err, &store.ResourceConflictError{}):
```

Comment:
1. It seems this case `errors.Is(err, &Conflict{})` is invalid, as `*Conflict` doesn't have `Is(err error) bool` method, so it will always return `false`

2. 
```suggestion
	case errors.Is(err, &Conflict{}), errors.Is(err, &store.ResourceConflictError{}):
```

3.
The same suggestion as mentioned earlier, I think this PR is a great opportunity to make other Errors consistent, so I would refactor [these](https://github.com/kumahq/kuma/blob/51bb6190f28e05c4d826372975f47221f92a62c1/pkg/core/rest/errors/errors.go#L3-L25):
```go
package errors

import (
	"reflect"
)

type UnauthenticatedError struct{}

func (e *UnauthenticatedError) Error() string {
	return "Unauthenticated"
}

func (e *UnauthenticatedError) Is(err error) bool {
	return reflect.TypeOf(err) == reflect.TypeOf(e)
}

type MethodNotAllowedError struct{}

func (e *MethodNotAllowedError) Error() string {
	return "Method not allowed"
}

func (e *MethodNotAllowedError) Is(err error) bool {
	return reflect.TypeOf(err) == reflect.TypeOf(e)
}

type ConflictError struct{}

func (e *ConflictError) Error() string {
	return "Conflict"
}

func (e *ConflictError) Is(err error) bool {
	return reflect.TypeOf(err) == reflect.TypeOf(e)
}

type ServiceUnavailableError struct{}

func (e *ServiceUnavailableError) Error() string {
	return "Service unavailable"
}

func (e *ServiceUnavailableError) Is(err error) bool {
	return reflect.TypeOf(err) == reflect.TypeOf(e)
}
```

Reply:
Let's be careful of scope again here.

Reply:
As for the `IsConflict` AFAIK it works because there's no variable part in Conflict so `==` will be true which is the default very of `Is`